api/engineapi/routes/contracts.py

```diff
@@ -0,0 +1,203 @@
+async def list_registered_contracts(
+    request: Request,
+    blockchain: Optional[str] = Query(None),
+    address: Optional[str] = Query(None),
+    contract_type: Optional[contracts_actions.ContractType] = Query(None),
+    limit: int = Query(10),
+    offset: Optional[int] = Query(None),
+    db_session: Session = Depends(db.yield_db_session),
+) -> List[data.RegisteredContract]:
+    """
+    Users can use this endpoint to look up the contracts they have registered against this API.
+    """
+    contracts = contracts_actions.lookup_registered_contracts(
+        db_session=db_session,
+        moonstream_user_id=request.state.user.id,
+        blockchain=blockchain,
+        address=address,
+        contract_type=contract_type,
+        limit=limit,
+        offset=offset,
+    )
+    return [
+        contracts_actions.render_registered_contract(contract) for contract in contracts
+    ]
+
+
+@app.post("/register", response_model=data.RegisteredContract)
+async def register_contract(
+    request: Request,
+    contract: data.RegisterContractRequest,
+    db_session: Session = Depends(db.yield_db_session),
+) -> data.RegisteredContract:
+    """
+    Allows users to register contracts.
+    """
+    try:
+        registered_contract = contracts_actions.register_contract(
+            db_session=db_session,
+            moonstream_user_id=request.state.user.id,
+            blockchain=contract.blockchain,
+            address=contract.address,
+            contract_type=contracts_actions.ContractType(contract.contract_type),
+            title=contract.title,
+            description=contract.description,
+            image_uri=contract.image_uri,
+        )
+    except contracts_actions.ContractAlreadyRegistered:
+        raise EngineHTTPException(
+            status_code=409,
+            detail="Contract already registered",
+        )
+    return contracts_actions.render_registered_contract(registered_contract)
+
+
+@app.delete("/{contract_id}", response_model=data.RegisteredContract)
+async def delete_contract(
+    request: Request,
+    contract_id: UUID,
+    db_session: Session = Depends(db.yield_db_session),
+) -> data.RegisteredContract:
+    """
+    Allows users to delete contracts that they have registered.
+    """
+    try:
+        deleted_contract = contracts_actions.delete_registered_contract(
+            db_session=db_session,
+            moonstream_user_id=request.state.user.id,
+            registered_contract_id=contract_id,
+        )
+    except Exception as err:
+        logger.error(repr(err))
+        raise EngineHTTPException(status_code=500)
+
+    return contracts_actions.render_registered_contract(deleted_contract)
+
+
+@app.get("/requests", response_model=List[data.CallRequest])
+async def list_requests(
+    contract_id: UUID = Query(...),
+    caller: str = Query(...),
+    limit: int = Query(100),
+    offset: Optional[int] = Query(None),
+    db_session: Session = Depends(db.yield_db_session),
```

Comment:
This is read handler, better to use read only database connection 
`db.yield_db_read_only_session`